CalendarApp/GoogleCalendarService/Manager/EventPeriodicityHolder.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using Google.Apis.Calendar.v3.Data;
+
+namespace GoogleCalendarService.Manager
+{
+    public class EventPeriodicityHolder
+    {
+        private readonly Event _realEvent;
+        private readonly string _frequency;
+        private List<DayOfWeek> _dayOfWeeksList;
+        private readonly int _count;
+        public List<Event> EventList;
+        private DateTime _startDateTime;
+        private DateTime _endDateTime;
+
+        public EventPeriodicityHolder(Event eventForAdd)
+        {
+            if (eventForAdd.Recurrence == null)
+            {
+                _dayOfWeeksList = new List<DayOfWeek>();
+                _dayOfWeeksList.Add(eventForAdd.Start.DateTime.Value.DayOfWeek);
+                _count = 1;
+                _frequency = "DAILY";
+            }
+            else
+            {
+                string mainRule = eventForAdd.Recurrence[0].Split(':')[1];
+                string[] rules = mainRule.Split(';');
+                foreach (var rule in rules)
+                {
+                    var ruleStr = rule.Split('=');
+                    if (ruleStr[0] == "FREQ")
+                        _frequency = ruleStr[1];
+                    else if (ruleStr[0] == "BYDAY")
+                        DayOfWeeksListSet(ruleStr[1]);
+                    else if (ruleStr[0] == "COUNT")
+                        _count = Convert.ToInt32(ruleStr[1]);
+                }
+            }
+            _startDateTime = eventForAdd.Start.DateTime.Value;
+            _endDateTime = eventForAdd.End.DateTime.Value;
+            _realEvent = eventForAdd;
+            CreateEventList();
+        }
+
+        private void DayOfWeeksListSet(string byDay)
+        {
+            _dayOfWeeksList = new List<DayOfWeek>();
+            string[] byDayStr = byDay.Split(',');
+            foreach (string day in byDayStr)
+            {
+                switch (day)
+                {
+                    case "SU": _dayOfWeeksList.Add(DayOfWeek.Sunday); continue;
+                    case "MO": _dayOfWeeksList.Add(DayOfWeek.Monday); continue;
+                    case "TU": _dayOfWeeksList.Add(DayOfWeek.Tuesday); continue;
+                    case "WE": _dayOfWeeksList.Add(DayOfWeek.Wednesday); continue;
+                    case "TH": _dayOfWeeksList.Add(DayOfWeek.Thursday); continue;
+                    case "FR": _dayOfWeeksList.Add(DayOfWeek.Friday); continue;
+                    case "SA": _dayOfWeeksList.Add(DayOfWeek.Saturday); continue;
```

Comment:
extract this switch into a method that returns DayOfWeek and replace this entire method with a single byDay.Split(',').Select(ExtractedMethod).ToList()